### PR TITLE
XML Encode the windows XML data

### LINKED ISSUE #3195: When packaging an ampersand in "Name" causes an error
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When packaging a Fyne app I cam across an issue where I had the application name in the `FyneApp.toml` containing an ampersand:

```toml
[Details]
  Icon = "logo.png"
  Name = "App With & In Name"
  ID = "com.example.app-with-ampersand-in-name"
  Version = "1.1.0"
  Build = 1
```

This resulted in an error `The application has failed to start because its side-by-side configuration is incorrect. Please see the application event log or use the command-line sxstrace.exe tool for 
more detail` similar to #1521

After reviewing the Windows Event Log there were entries showing `Error in manifest or policy file "C:\path\to\app.exe" on line 2. Invalid Xml syntax`

After replacing the `&` with `and` the issue went away.

I would assume the name is not being properly escaped in the generated manifest as an ampersand must be escaped as `&amp;` in XML.

### How to reproduce

1. Create `FyneApp.toml` with `Name` containing and ampersand character
2. Package using `fyne package -os windows`
3. Attempt to run app

### Screenshots

_No response_

### Example code

No code example, issue is with app manifest generation of `fyne` command line tool. 

### Fyne version

2.2.3

### Go compiler version

1.19

### Operating system

Windows

### Operating system version

Windows 11

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by andydotxyz**
Fixes #3195

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: cmd/fyne/internal/commands/encode.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package commands
+
+import "strings"
+
+func encodeXMLString(in string) string {
+	amped := strings.ReplaceAll(in, "&", "&amp;")
+	return strings.ReplaceAll(strings.ReplaceAll(amped, "<", "&lt;"), ">", "&gt;")
```
Comment (Jacalz) [inline cmd/fyne/internal/commands/encode.go:7]: I wonder if there perhaps isn't a better alternative already in the standard library?
[This link](https://stackoverflow.com/questions/22906722/how-to-encode-special-characters-in-xml) seems to indicate that there are quite a few special characters.

Some googling on "html escape string in Go" seems to bring this up: https://stackoverflow.com/questions/38065765/how-to-escape-the-html-using-golang.

Comment (Jacalz) [reply cmd/fyne/internal/commands/encode.go:7]: Using [html.EscapeString](https://pkg.go.dev/html#EscapeString) seems to be the right way to go.

File: cmd/fyne/internal/commands/encode.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package commands
+
+import (
+	"html"
+)
+
+func encodeXMLString(in string) string {
+	return html.EscapeString(in)
```
Comment (Jacalz) [inline cmd/fyne/internal/commands/encode.go:8]: I don't think we need the wrapper function nor the tests if we are to use the built in function.

**Review by Jacalz — CHANGES_REQUESTED** [review]

File: cmd/fyne/internal/commands/encode.go (hunk 2 above)
Comment (andydotxyz) [reply cmd/fyne/internal/commands/encode.go:8]: Given that HTML and XML are not actually the same I felt the clarity was helpful.

Comment (Jacalz) [reply cmd/fyne/internal/commands/encode.go:8]: Good point

**Review by Jacalz — APPROVED** [review]

**Pull request merged**